Document/0x06b-Basic-Security-Testing.md

```diff
@@ -786,42 +757,298 @@ Possible values for the property [UIDeviceFamily](https://developer.apple.com/li
+
+  ```bash
+  $ plutil -convert xml1 Info.plist
+  ```
+
+- On Linux:
+
+  ```bash
+  $ apt install libplist-utils
+  $ plistutil -i Info.plist -o Info_xml.plist
+  ```
+
+Here's a non-exhaustive list of some info and the corresponding keywords that you can easily search for in the `Info.plist` file by just inspecting the file or by using `grep -i <keyword> Info.plist`:
+
+- App permissions Purpose Strings: `UsageDescription` (see "iOS Platform APIs")
+- Custom URL schemes: `CFBundleURLTypes` (see "iOS Platform APIs")
+- Exported/imported *custom document types*: `UTExportedTypeDeclarations`/`UTImportedTypeDeclarations` (see "iOS Platform APIs")
+- App Transport Security (ATS) configuration: `NSAppTransportSecurity` (see "iOS Network APIs")
+
+Please refer to the mentioned chapters to learn more about how to test each of these points.
+
+###### App Binary
+
+iOS app binaries are fat binaries (they can be deployed on all devices 32- and 64-bit). In contrast to Android, where you can actually decompile the app binary to Java code, the iOS app binaries can only be disassembled.
+
+Refer to the chapter "Reverse Engineering and Tampering on iOS" for more details.
+
+###### Native Libraries
+
+iOS native libraries are known as Frameworks. You can inspect them by listing the `Frameworks` folder in the IPA:
+
+```shell
+$ ls -1 Frameworks/
+Realm.framework
+libswiftCore.dylib
+libswiftCoreData.dylib
+libswiftCoreFoundation.dylib
+```
 
-Once you have installed the app, there is further information to explore, where tools like objection come in handy. Connecting to the application with objection is described in the section "Recommended Tools - Objection".
+or from the device with objection (as well as per SSH of course):
 
-When using objection you can retrieve different kinds of information, where `env` will show you all the directory information of iGoat.
+```shell
+OWASP.iGoat-Swift on (iPhone: 11.1.2) [usb] # ls
+NSFileType      Perms  NSFileProtection    ...  Name
+------------  -------  ------------------  ...  ----------------------------
+Directory         493  None                ...  Realm.framework
+Regular           420  None                ...  libswiftCore.dylib
+Regular           420  None                ...  libswiftCoreData.dylib
+Regular           420  None                ...  libswiftCoreFoundation.dylib
+...
+```
+
+For now this is all information you can get about the Frameworks unless you start reverse engineering them. Refer to the chapter "Tampering and Reverse Engineering on iOS" for more information about how to reverse engineer Frameworks.
+
+###### Other App Resources
+
+It is normally worth taking a look at the rest of the resources and files that you may find in the Application Bundle (.app) inside the IPA as some times they contain additional goodies like encrypted databases, certificates, etc.
```

Comment:
I leave it like it is because:

> if "etc." is used in the middle of a sentence, it is followed by a comma. (Tennis, soccer, baseball, etc., are outdoor games.) However, if this word appears at the end of a sentence then the period (which is part of "etc.") serves as the final punctuation mark.